src/gg20/crypto_tools/paillier/zk/composite_dlog.rs

```diff
@@ -30,30 +30,41 @@ use super::{super::utils::member_of_mul_group, NIZKStatement};
 
 #[derive(Debug, Clone, PartialEq, Serialize, Deserialize, Zeroize)]
 pub struct CompositeDLogStmt {
+    // TODO no need for `pub` if we move things around a bit
     pub n: BigNumber,
     pub g: BigNumber,
     pub v: BigNumber,
 }
 
 #[derive(Debug, Clone, PartialEq, Serialize, Deserialize, Zeroize)]
 pub struct CompositeDLogProof {
-    pub(crate) x: BigNumber,
-    pub(crate) y: BigNumber,
+    x: BigNumber,
+    y: BigNumber,
 }
 
 pub struct CompositeDLogWitness {
-    pub(crate) s: Randomness,
-    pub(crate) size: usize,
+    s: Randomness,
+    size: usize,
 }
 
 // The challenge size is likely a conservative choice as opposed to 128.
 // This was chosen since we used SHA256 to compute
 // the challenge hash which is 256 bits long.
 const CHALLENGE_K: usize = 256;
 const SECURITY_PARAM_K_PRIME: usize = 128;
-const WITNESS_SIZE: usize = 256;
-/// The max witness size is `max(witness_size, modulus_max_size) = modulus_max_size`
-const MAX_R_SIZE: usize = CHALLENGE_K + SECURITY_PARAM_K_PRIME + MODULUS_MAX_SIZE;
+const S_WITNESS_SIZE: usize = 256;
+
+/// s^-1 is the inverse of the S_WITNESS_SIZE-bit number s modulo a MODULUS_MAX_SIZE-bit modulus,
+/// so the size of s^-1 is MODULUS_MAX_SIZE
+const S_INV_WITNESS_SIZE: usize = MODULUS_MAX_SIZE;
+
+/// `r` masks a witness `s`.  `MAX_R_MASK_SIZE` is the size of the largest possible such mask.
+const MAX_R_MASK_SIZE: usize = r_mask_size(max(S_WITNESS_SIZE, S_INV_WITNESS_SIZE));
```

Comment:
Thanks, I tried using `max` but it was complaining about not being a `const` function. It seems I was using the `std::cmp::max` function which is different than the one already in scope.